dotnet/src/AutoGen.Mistral/Agent/MistralClientAgent.cs

```diff
@@ -0,0 +1,106 @@
+
+public class MistralClientAgent : IStreamingAgent
+{
+    private readonly MistralClient _client;
+    private readonly string _systemMessage;
+    private readonly string _model;
+    private readonly int? _randomSeed;
+    private readonly bool _jsonOutput = false;
+    private ToolChoiceEnum? _toolChoice;
+    public MistralClientAgent(
+        MistralClient client,
+        string name,
+        string model,
+        string systemMessage = "You are a helpful AI assistant",
+        int? randomSeed = null,
+        ToolChoiceEnum? toolChoice = null,
+        bool jsonOutput = false)
+    {
+        _client = client;
+        Name = name;
+        _systemMessage = systemMessage;
+        _model = model;
+        _randomSeed = randomSeed;
+        _jsonOutput = jsonOutput;
+        _toolChoice = toolChoice;
+    }
+
+    public string Name { get; }
+
+    public async Task<IMessage> GenerateReplyAsync(
+        IEnumerable<IMessage> messages,
+        GenerateReplyOptions? options = null,
+        CancellationToken cancellationToken = default)
+    {
+        var request = BuildChatRequest(messages, options);
+        var response = await _client.CreateChatCompletionsAsync(request);
+
+        return new MessageEnvelope<ChatCompletionResponse>(response, from: this.Name);
+    }
+
+    public async Task<IAsyncEnumerable<IStreamingMessage>> GenerateStreamingReplyAsync(
+        IEnumerable<IMessage> messages,
+        GenerateReplyOptions? options = null,
+        CancellationToken cancellationToken = default)
+    {
+        var request = BuildChatRequest(messages, options);
+        var response = _client.StreamingChatCompletionsAsync(request);
+
+        return ProcessMessage(response);
+    }
+
+    private async IAsyncEnumerable<IMessage> ProcessMessage(IAsyncEnumerable<ChatCompletionResponse> response)
+    {
+        await foreach (var content in response)
+        {
+            yield return new MessageEnvelope<ChatCompletionResponse>(content, from: this.Name);
+        }
+    }
+
+    private ChatCompletionRequest BuildChatRequest(IEnumerable<IMessage> messages, GenerateReplyOptions? options)
+    {
+        var chatHistory = BuildChatHistory(messages);
+        var chatRequest = new ChatCompletionRequest(model: _model, messages: chatHistory.ToList(), temperature: options?.Temperature, randomSeed: _randomSeed)
+        {
+            MaxTokens = options?.MaxToken,
+            ResponseFormat = _jsonOutput ? new ResponseFormat() { ResponseFormatType = "json_object" } : null,
+        };
+
+        if (options?.Functions != null)
+        {
+            chatRequest.Tools = options.Functions.Select(f => new FunctionTool(f.ToMistralFunctionDefinition())).ToList();
+            chatRequest.ToolChoice = _toolChoice ?? ToolChoiceEnum.Auto;
+        }
+
+        return chatRequest;
+    }
+
+    private IEnumerable<ChatMessage> BuildChatHistory(IEnumerable<IMessage> messages)
+    {
+        var history = messages.Select(m => m switch
```

Comment:
Is this a little unexpected for users if both _systemMessage is set and system messages exist in the chat history?  I have encountered scenarios where task-specific system message instructions are needed. e.g., in a sequence of different chats with the same agent, each chat has different instructions.

Maybe the more intuitive behavior is to have both _systemMessage and the input system messages in the history.

Reply:
It is possible to have multiple system messages, so some usage cases will take advantage of this. For now, we can make the behavior consistent with existing agent classes. We can make changes in the future the usage cases for .NET arises.